### PR TITLE
Optionals

### PR DESCRIPTION
**Pull request by tiagogomez**

### CODE REVIEW

File: Optionals.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
+//: Playground - noun: a place where people can play
+
+import Foundation
+import XCTest
+
+struct OptionalElement {
+    var optionalName: String?
+    var optionalNumber: Int?
+    var optionalPropertie: OptionalPropertie?
```
Comment (josoriora) [inline Optionals.playground/Contents.swift:9]: typo: it is `optionalProperty`

File: Optionals.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+struct OptionalPropertie {
```
Comment (josoriora) [inline Optionals.playground/Contents.swift:12]: typo: it is `optionalProperty`

File: Optionals.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    
+    func testForcedUnwrap(){
+        let expectedValue = "Desenvuelto"
+        let optionalValue: String? = "Desenvuelto"
+        if optionalValue != nil{
+            let unwrappedValue = optionalValue!     // Never force unwrap an optional that’s nil
+            XCTAssertEqual(unwrappedValue, expectedValue, "The optional value should be unwrapped")
+        }else {
+            XCTAssertTrue(false, "The element is not being unwrapped")
```
Comment (josoriora) [inline Optionals.playground/Contents.swift:44]: It is better to user `XCTFail()` in these cases.

File: Optionals.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    }
+    
+    func testOptionalBindingToUnwrap(){
+        let expectedValue = "Desenvuelto"
+        let optionalValue: String? = "Desenvuelto"
+        if let unwrappedValue = optionalValue {
+            XCTAssertEqual(unwrappedValue, expectedValue, "The optional value should be unwrapped")
+        }else {
+            XCTAssertTrue(false, "The element is not being unwrapped")
```
Comment (josoriora) [inline Optionals.playground/Contents.swift:54]: It is better to user `XCTFail()` in these cases.

File: Optionals.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        let expectedValue2: String = "Años"
+        let optionalValue1: Int? = 10
+        let optionalValue2: String? = "Años"
+        if let unwrappedValue1 = optionalValue1,
+           let unwrappedValue2 = optionalValue2
+        {
+            XCTAssertEqual(unwrappedValue1, expectedValue1, "The optional value should be unwrapped")
+            XCTAssertEqual(unwrappedValue2, expectedValue2, "The optional value should be unwrapped")
+        }else {
+            XCTAssertTrue(false, "The elements are not being unwrapped")
```
Comment (josoriora) [inline Optionals.playground/Contents.swift:69]: It is better to user `XCTFail()` in these cases.

File: Optionals.playground/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    }
+    
+    func testOptionalChainingToUnwrap() {
+        let expectedElement = OptionalPropertie(optionalText: "NotEmpty")
+        var optionalElement = OptionalElement(optionalName: nil, optionalNumber: nil, optionalPropertie: nil)
+        optionalElement.optionalPropertie?.optionalText = "NotEmpty"    //This line shold not work because The optionalPropertie is nil
+        XCTAssertNotEqual(optionalElement.optionalPropertie?.optionalText, expectedElement.optionalText, "The optional value should be nil")
+        let optionalPropertie = OptionalPropertie()
```
Comment (josoriora) [inline Optionals.playground/Contents.swift:78]: typo: It is `optionalProperty`

File: Optionals.playground/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    var optionalName: String?
+    var optionalNumber: Int?
+    var optionalPropertie: OptionalPropertie?
+}
+
+struct OptionalPropertie {
+    var optionalText: String?
+}
+
+class Optionals: XCTestCase{
```
Comment (josoriora) [inline Optionals.playground/Contents.swift:16]: I want a test that has a function that returns an optional also.

Comment (josoriora) [reply Optionals.playground/Contents.swift:16]: I also would like a test that expresses that the short form and long form of the optional are equivalent using `some` and `none`

let shortForm: Int? = Int("42")
let longForm: Optional<Int> = Int("42")

let number: Int? = Optional.some(42)
let noNumber: Int? = Optional.none
